search/facets_builder.go

```diff
@@ -54,23 +54,22 @@ type FacetBuilder interface {
 
 type FacetsBuilder struct {
 	indexReader index.IndexReader
-	facets      map[string]FacetBuilder
+	facetNames  []string
+	facets      []FacetBuilder
 	fields      []string
 }
 
 func NewFacetsBuilder(indexReader index.IndexReader) *FacetsBuilder {
 	return &FacetsBuilder{
 		indexReader: indexReader,
-		facets:      make(map[string]FacetBuilder, 0),
 	}
 }
 
 func (fb *FacetsBuilder) Size() int {
 	sizeInBytes := reflectStaticSizeFacetsBuilder + size.SizeOfPtr
 
 	for k, v := range fb.facets {
-		sizeInBytes += size.SizeOfString + len(k) +
-			v.Size()
+		sizeInBytes += size.SizeOfString + v.Size() + len(fb.facetNames[k])
```

Comment:
Actually you dont need this extra SizeOfString anymore.

Reply:
My thinking was that, size.SizeOfString needs to be counted for the new facetNames[] string slice and just reused the same existing facets size count loop for doing the same. 
size.SizeOfString + len(fb.facetNames[k] => for the new facetNames slice 

But think I missed something here,
Now for accounting the fb.facets correctly,  need to add a 
size.SizeOfUint64 along with this, right?

Reply:
Ah you're right, my mistake.